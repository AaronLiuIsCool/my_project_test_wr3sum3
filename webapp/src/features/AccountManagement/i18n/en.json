{
    "heading": "账号管理",
    "basicInfo": "基本信息",
    "accountManagement": "账号管理",
    "wechat": "微信",
    "email": "邮箱",
    "password": "密码",
    "bind": "前往绑定",
    "unbind": "解除绑定"
  }