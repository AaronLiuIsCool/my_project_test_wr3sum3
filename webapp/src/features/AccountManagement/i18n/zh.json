{
  "heading": "账号管理",
  "basicInfo": "基本信息",
  "accountManagement": "账号管理",
  "wechat": "微信",
  "email": "邮箱",
  "password": "密码",
  "bind": "前往绑定",
  "unbind": "解除绑定",
  "binded": "已绑定",
  "notBinded": "未绑定",
  "reset": "重置",
  "enterEmailAddress": "请输入您所要绑定的邮箱",
  "save": "保存",
  "avatar": "头像",
  "name": "姓名",
  "updateEmail": "更换邮箱",
  "gender": "性别",
  "identity": "身份",
  "genderOptions": [
    {
      "key": "male",
      "value": "男"
    },
    {
      "key": "female",
      "value": "女"
    }
  ],
  "identityOptions": [
    {
      "key": "male",
      "value": "男"
    },
    {
      "key": "female",
      "value": "女"
    }
  ]
}
