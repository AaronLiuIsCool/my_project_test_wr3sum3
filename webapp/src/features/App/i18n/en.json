{
  "brand": "Tailoryourvc",

  "nav_item_resumehub": "Resume Hub",
  "nav_item_jobmatcher": "Jobs",

  "language_toggle_zh": "中文",
  "language_toggle_en": "English",

  "logout": "Sign out",
  "defaultInitial": "Unknown"
}