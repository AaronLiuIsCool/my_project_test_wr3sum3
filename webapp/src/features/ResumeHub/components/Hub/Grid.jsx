import React from 'react';
import PropTypes from 'prop-types';

import styles from '../../styles/Hub.module.css';

const Grid = ({ children }) => (
    <div className={styles.grid}>
        {children}
    </div>
);

Grid.propTypes = {
    children: PropTypes.arrayOf(PropTypes.element)
};

export default Grid;