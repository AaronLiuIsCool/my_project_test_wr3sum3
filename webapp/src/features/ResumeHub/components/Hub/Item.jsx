import React from 'react';
import PropTypes from 'prop-types';
import { useSelector, useDispatch } from 'react-redux';
import { Link } from "react-router-dom";

import Button from 'react-bootstrap/Button';
import thumbnail from '../../assets/thumbnail@2x.png'; 
import { ReactComponent as EditIcon } from '../../assets/edit.svg'; 
import { ReactComponent as Download } from '../../assets/download.svg'; 
import { ReactComponent as ShareIcon } from '../../assets/share.svg'; 
import { ReactComponent as DuplicateIcon } from '../../assets/duplicate.svg'; 
import { ReactComponent as DeleteIcon } from '../../assets/delete.svg'; 

import { selectUserId } from 'features/App/slicer';
import { deleteResume } from 'features/App/slicer/account';
import AccountServices from 'shell/services/AccountServices';
import { getLogger } from 'shell/logger';

import styles from '../../styles/Hub.module.css';

const logger = getLogger('ResumeHubItem');
const accountServices = new AccountServices();

const Grid = ({ resume }) => {
    const dispatch = useDispatch();
    const userId = useSelector(selectUserId);

    const handleDelete = async () => {
        dispatch(deleteResume(resume.resumeId));
        try {
            const response = await accountServices.deleteResume(userId, resume);
            const responseJson = await response.json();
            if (!response.success) {
                logger.warn(responseJson.message);
            }
        } catch (exception) {
            logger.error(exception)
        }
    }

    return (
        <div className={styles.itemContainer}>
            <div className={styles.itemContent}>
                <img src={thumbnail} width={160} height={208} alt="Thumbnail" />
                <div className={styles.right}>
                    <div className={styles.name}>
                        {resume.alias}
                    </div>
                    <div className={styles.time}>
                        {new Date(resume.createdAt).toLocaleDateString()}
                    </div>
                    <div className={styles.links}>
                        <Link to={`/resume/${resume.resumeId}`} className={styles.link} >
                            <EditIcon /> 编辑内容
                        </Link>
                        <Button variant="link" className={styles.link}>
                            <Download /> 下载简历
                        </Button>
                        <Button variant="link" className={styles.link}>
                            <ShareIcon /> 链接分享
                        </Button>
                        <Button variant="link" className={styles.link}>
                            <DuplicateIcon /> 复制简历
                        </Button>
                    </div>
                    <Button variant="link" className={styles.delete}
                        onClick={handleDelete} >
                        <DeleteIcon /> 删除
                    </Button>
                </div>
            </div>
            <Button className={styles.matchingBtn}>
                工作匹配
            </Button>
        </div>
    );
}

Grid.propTypes = {
    resume: PropTypes.shape({
        resumeId: PropTypes.string.isRequired,
        alias: PropTypes.string.isRequired,
        thumbnailUri: PropTypes.string.isRequired,
        createdAt: PropTypes.string.isRequired
    })
};

export default Grid;