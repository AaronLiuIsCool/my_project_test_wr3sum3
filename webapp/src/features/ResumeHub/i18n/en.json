{
  "new_resume_description": "You do not have a resume yet, spend five minutes to creat one?",
  "new_resume_start": "Build resume"
}