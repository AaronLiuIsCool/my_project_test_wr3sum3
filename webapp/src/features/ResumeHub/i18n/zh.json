{
  "new_resume_description": "看似您还没有简历，花5分钟创造一份吧",
  "new_resume_start": "开始创建",

  "hub_title": "我的简历",
  "hub_action": "+ 添加新简历",

  "hub_item_edit": "编辑内容",
  "hub_item_download": "下载简历",
  "hub_item_share": "链接分享",
  "hub_item_duplicate": "复制简历",
  "hub_item_delete": "删除",
  "hub_item_match": "工作匹配"
}