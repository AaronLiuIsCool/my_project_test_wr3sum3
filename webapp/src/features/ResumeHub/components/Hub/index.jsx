import React from 'react';
import PropTypes from 'prop-types';

import Button from 'react-bootstrap/Button';
import { Link } from 'react-router-dom'
import Grid from './Grid';
import Item from './Item';

import { useI8n } from 'shell/i18n';

import styles from '../../styles/Hub.module.css';

function renderGrid(resumes, account) {
    const resumeItems = resumes.map(resume => <Item account={account} resume={resume} key={resume.resumeId} />);
    return <Grid>{resumeItems}</Grid>;
}

const Hub = ({ resumes, account }) => {
    const messages = useI8n();
    return (
        <div className={styles.container}>
            <div className={styles.header}>
                
            <Button as={Link} variant="link" className={styles.action} to='/new'>
                {messages['hub_action']}
            </Button>
            </div>
            {renderGrid(resumes, account)}
        </div>
    );
}

Hub.propTypes = {
    resumes: PropTypes.arrayOf(PropTypes.shape({
        resumeId: PropTypes.string.isRequired,
        alias: PropTypes.string.isRequired,
        thumbnailUri: PropTypes.string.isRequired,
        createdAt: PropTypes.string.isRequired
    }))
};

export default Hub;