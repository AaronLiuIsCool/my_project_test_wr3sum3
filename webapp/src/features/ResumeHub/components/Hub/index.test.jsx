import React from 'react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';

import store from 'store';
import { I8nContext } from 'shell/i18n';

import Hub from './index';

const mockData = [{
    "resumeId":"4028b401755f29230175614025980000",
    "alias":"resume 1 for test",
    "thumbnailUri":"thumbnailUri",
    "createdAt":"2020-10-25T19:33:16Z"
}, {
    "resumeId":"4028b401755f292301756311ef930004",
    "alias":"test 2",
    "thumbnailUri":"thumbnailUri",
    "createdAt":"2020-10-26T04:02:01Z"
}];

describe('Hub', () => {
    let dateSpy;
    beforeEach(() => {
        const dateMock = new Date();
        dateMock.toLocaleString = jest.fn(() => 'test locale date string');
        dateSpy = jest.spyOn(global, 'Date')
                      .mockImplementation(() => dateMock);
        dateSpy.now = jest.fn(() => 5454684654564);
    });

    afterEach(() => {
        dateSpy.mockRestore();
    });

    test('renders Hub', () => {
        const hub = renderer.create(
            <BrowserRouter>
                <Provider store={store}>
                    <I8nContext.Provider value={{}}>
                        <Hub resumes={mockData} />
                    </I8nContext.Provider>
                </Provider>
            </BrowserRouter>
        ).toJSON();
    
        expect(hub).toMatchSnapshot();
    });    
});
