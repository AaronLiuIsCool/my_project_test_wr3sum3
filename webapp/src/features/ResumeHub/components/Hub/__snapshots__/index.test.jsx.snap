// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hub renders Hub 1`] = `
<div
  className="container"
>
  <div
    className="header"
  >
    <a
      className="action btn btn-link"
      disabled={false}
      href="/new"
      onClick={[Function]}
    />
  </div>
  <div
    className="grid"
  >
    <div
      className="itemContainer"
    >
      <div
        className="itemContent"
      >
        <div
          className="right"
        >
          <div
            className="aliasContainer"
          >
            <input
              className="name"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              value=""
            />
            <div
              className="edit"
            >
              <img
                alt="edit"
                src="edit.svg"
              />
            </div>
          </div>
          <div
            className="time"
          >
            2018–01–30T12:34:56+00:00
          </div>
          <div
            className="links"
          >
            <a
              className="link btn btn-link"
              disabled={false}
              href="/resume/4028b401755f29230175614025980000"
              onClick={[Function]}
            >
              <svg
                className="svg"
              >
                edit.svg
              </svg>
            </a>
            <button
              className="link btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                className="svg"
              >
                download.svg
              </svg>
            </button>
          </div>
          <button
            className="delete btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              className="svg"
            >
              delete.svg
            </svg>
             
          </button>
        </div>
      </div>
      <a
        className="matchingBtn btn btn-primary"
        disabled={false}
        href="/jobs?resume=4028b401755f29230175614025980000"
        onClick={[Function]}
      />
    </div>
    <div
      className="itemContainer"
    >
      <div
        className="itemContent"
      >
        <div
          className="right"
        >
          <div
            className="aliasContainer"
          >
            <input
              className="name"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              value=""
            />
            <div
              className="edit"
            >
              <img
                alt="edit"
                src="edit.svg"
              />
            </div>
          </div>
          <div
            className="time"
          >
            2018–01–30T12:34:56+00:00
          </div>
          <div
            className="links"
          >
            <a
              className="link btn btn-link"
              disabled={false}
              href="/resume/4028b401755f292301756311ef930004"
              onClick={[Function]}
            >
              <svg
                className="svg"
              >
                edit.svg
              </svg>
            </a>
            <button
              className="link btn btn-link"
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                className="svg"
              >
                download.svg
              </svg>
            </button>
          </div>
          <button
            className="delete btn btn-link"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            <svg
              className="svg"
            >
              delete.svg
            </svg>
             
          </button>
        </div>
      </div>
      <a
        className="matchingBtn btn btn-primary"
        disabled={false}
        href="/jobs?resume=4028b401755f292301756311ef930004"
        onClick={[Function]}
      />
    </div>
  </div>
</div>
`;
